src/charm.py

```diff
@@ -214,6 +217,17 @@ def _on_collect_unit_status(self, event: ops.CollectStatusEvent) -> None:
         if not self.model.relations.get("cos-agent"):
             event.add_status(BlockedStatus("Grafana Agent is not related"))
 
+        upstream_snap = get_installed_snap_service(UPSTREAM_SNAP)
+
+        # this is necessary when doing a charm upgrade coming from revision 27
+        if upstream_snap.present:
+            event.add_status(
+                BlockedStatus(
+                    "Please add an empty file as resource. For more info see: "
+                    "https://charmhub.io/openstack-exporter#known-issues"
```

Comment:
We can link to the new tutorial here:
```suggestion
                    "Upgrades from revisions older than 28 require manual intervention. Please see: "
                    "https://charmhub.io/openstack-exporter/docs/upgrade-from-older-than-28"
```

Reply:
if the original message fits but the proposed one does not I'm fine with keeping the one that fits

Reply:
It does not fit:
```
Model  Controller  Cloud/Region         Version  SLA          Timestamp
test   lxd         localhost/localhost  3.4.5    unsupported  18:05:28-03:00

App                    Version          Status   Scale  Charm               Channel        Rev  Exposed  Message
grafana-agent                           blocked      1  grafana-agent       latest/stable  216  no       Missing ['grafana-cloud-config']|['grafana-dashboards-provider']|['logging-consumer']|['send-remote-write'] for cos-a...
keystone               21.0.1           active       1  keystone            yoga/stable    689  no       Application Ready
keystone-mysql-router  8.0.39           active       1  mysql-router        8.0/stable     200  no       Unit is ready
mysql                  8.0.36-0ubun...  active       1  mysql               8.0/stable     240  no       
openstack-exporter                      blocked      1  openstack-exporter                  11  no       Upgrades from revisions older than 28 require manual intervention. Please see https://charmhub.io/openstack-exporter/...

Unit                        Workload  Agent  Machine  Public address  Ports           Message
keystone/0*                 active    idle   1        10.101.82.139   5000/tcp        Unit is ready
  keystone-mysql-router/0*  active    idle            10.101.82.139                   Unit is ready
mysql/0*                    active    idle   2        10.101.82.220   3306,33060/tcp  Primary
openstack-exporter/0*       blocked   idle   0        10.101.82.235                   Upgrades from revisions older than 28 require manual intervention. Please see https://charmhub.io/openstack-exporter/...
  grafana-agent/0*          blocked   idle            10.101.82.235                   Missing ['grafana-cloud-config']|['grafana-dashboards-provider']|['logging-consumer']|['send-remote-write'] for cos-a...

Machine  State    Address        Inst id        Base          AZ  Message
0        started  10.101.82.235  juju-966319-0  ubuntu@22.04      Running
1        started  10.101.82.139  juju-966319-1  ubuntu@22.04      Running
2        started  10.101.82.220  juju-966319-2  ubuntu@22.04      Running
```

Reply:
How about
```suggestion
                    "golang-openstack-exporter detected. Please see: "
                    "https://charmhub.io/openstack-exporter#known-issues"
```